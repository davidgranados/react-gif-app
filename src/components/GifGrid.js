import React from "react";
import PropTypes from "prop-types";
import GifGridItem from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

const GifGrid = ({ category }) => {
  const { data: gifs, loading } = useFetchGifs(category);
  return (
    <>
      <h3>{category}</h3>
      {loading && <p className="animate__animated animate__flash">Cargando...</p>}
      {!!gifs.length && (
        <div className="card-grid animate__animated animate__fadeIn">
          {gifs.map((gif) => (
            <GifGridItem key={gif.id} {...gif} />
          ))}
        </div>
      )}
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
