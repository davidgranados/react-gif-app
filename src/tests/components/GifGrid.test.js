import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Tests on GifGrid Component", function () {
  const category = "marvel";
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it("should render correctly", function () {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should show items if useFetchGifs return objects", function () {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/cualquier/cosa1.jpg",
        title: "Cualquier cosa 1",
      },
      {
        id: "123",
        url: "https://localhost/cualquier/cosa2.jpg",
        title: "Cualquier cosa 2",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow( <GifGrid category={ category } /> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
  });
});
