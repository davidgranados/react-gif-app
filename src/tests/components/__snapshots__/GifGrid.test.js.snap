// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifGrid Component should render correctly 1`] = `
<Fragment>
  <h3>
    marvel
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
</Fragment>
`;

exports[`Tests on GifGrid Component should show items if useFetchGifs return objects 1`] = `
<Fragment>
  <h3>
    marvel
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa 1"
      url="https://localhost/cualquier/cosa1.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa 2"
      url="https://localhost/cualquier/cosa2.jpg"
    />
  </div>
</Fragment>
`;
